### PR TITLE
docs: Improve Kafka client tutorials

### LINKED ISSUE #1367: Use context switcher for quickstart options
### Describe the Issue

The current quickstart displays tabs under each heading: one for Redpanda Cloud and one for Docker.
To allow Cloud users to find this topic in the nav tree, we link to it from the **Deploy** > **Cloud** menu item. However, we cannot link to specific tabs, so Cloud users often see the Docker instructions by default.

To make the experience better for each persona (Docker/Cloud), we should use our new context switcher UI.

### Updates to existing documentation

https://docs.redpanda.com/docs/get-started/quick-start/

### LINKED ISSUE #1411: single source NodeJS example and do similar cleanup for Go and Python examples
### Describe the Issue

The Node JS sample is duplicated 3 times in our doc set:
- https://docs.redpanda.com/docs/develop/guide-nodejs/
- https://docs.redpanda.com/docs/get-started/code-examples/
- https://docs.redpanda.com/docs/develop/code-examples/

The version that now is correct with PR #1402 is https://docs.redpanda.com/docs/develop/guide-nodejs/. 

Suggest putting the single source for this correct content into https://docs.redpanda.com/docs/get-started/code-examples/ and removing the other 2 pages with redirects. (Note that the 23.2 roadmap has an item for building a new advanced sample application.)

The Go and Python examples on https://docs.redpanda.com/docs/get-started/code-examples/ need to be tested and updated similar to way the NodeJS example was updated, so content can be easily copy/pasted. 

### Updates to existing documentation

- https://docs.redpanda.com/docs/develop/guide-nodejs/
- https://docs.redpanda.com/docs/get-started/code-examples/
- https://docs.redpanda.com/docs/develop/code-examples/

### LINKED ISSUE #1490: Update and improve overall quality of the Kafka Client libraries doc
https://docs.redpanda.com/docs/develop/code-examples/

The use cases in this doc are not quite sufficient for our customers. We have gotten feedback that they do not reflect "real world" customer contexts. Please work with eng and CS to pull in some examples that are more aligned with typical day 1 use cases.

Additionally, there seems to be some duplication of content between https://docs.redpanda.com/docs/develop/code-examples/ and https://docs.redpanda.com/docs/develop/guide-nodejs/. Please streamline this NodeJS content and remove any superfluous duplication.

### PR DESCRIPTION
**Pull request by JakeSCahill**
Review deadline: May 16

- Added more realistic code for Python and Go to mimic the chat room app in the Node.js tutorial.
- Removed the expandable example output from the quickstarts.
- Added download links for the `docker-compose.yml` files in the Docker quickstart.
- Fixed a bug in the rpk install docs that assumed users had access to the rpk binary before installing it.
- Split the quickstart into two pages: One for Cloud and one for Docker to make the content simpler to navigate.

Fixes #1490 fixes #1367 fixes #1411

### CODE REVIEW

**Review by michael-redpanda — COMMENTED** [review]
Main issue is that the NodeJS example does not build (at least for me).  Otherwise looks like a great update!

**Review by michael-redpanda — COMMENTED** [review]
Main issue is that the NodeJS example does not build (at least for me).  Otherwise looks like a great update!

File: docs/develop/guide-nodejs-cloud.mdx [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+
+<Intro/>
+
+## Prerequisites
+
+[Install Node.js for your platform](https://nodejs.org/en/download/package-manager/).
+
+## Set up Redpanda Cloud
+
+<SetUpCloud frontmatter={frontMatter}/>
```
[the remark this answers is not in the document]
Comment (michael-redpanda) [reply docs/develop/guide-nodejs-cloud.mdx:36]: I think my suggested change addresses this

File: docs/develop/guide-nodejs.mdx [hunk 2]
```diff
@@ -65,122 +63,131 @@ might affect your experience with other applications running on your machine.
...
-  })
+  brokers: ["localhost:19092"],
+});
+const admin = redpanda.admin();
+
+export async function createTopic(topic: string, partitions?: number, replicas?: number) {
+  try {
+    await admin.connect();
+
+    const { topics } = await admin.listTopics();
```
[the remark this answers is not in the document]
Comment (JakeSCahill) [reply docs/develop/guide-nodejs.mdx]: This should be fixed now with

```js
export async function createTopic(
  topic: string,
  partitions?: number,
  replicas?: number
) {
  await admin.connect();
  const existingTopics = await admin.listTopics();

  if (!existingTopics.includes(topic)) {
    await admin.createTopics({
      topics: [
        {
          topic: topic,
          numPartitions: partitions ? partitions : 1,
          replicationFactor: replicas ? replicas : 1,
        },
      ],
    });
  }

  await admin.disconnect();
}
```

File: docs/develop/guide-go-cloud.mdx [hunk 3]
```diff
@@ -16,7 +16,7 @@ contextLinks:
 </head>
 
 import Intro from "../develop/shared/chat-app/_intro.mdx"
-import SetUpDocker from "../develop/shared/chat-app/_set-up-docker.mdx"
+import SetUpCloud from "../develop/shared/chat-app/_set-up-docker.mdx"
```
Comment (michael-redpanda) [inline docs/develop/guide-go-cloud.mdx:19]: Wrong path here
```suggestion
import SetUpCloud from "../develop/shared/chat-app/_set-up-cloud.mdx"
```

File: docs/develop/guide-python-cloud.mdx [hunk 4]
```diff
@@ -16,7 +16,7 @@ contextLinks:
 </head>
 
 import Intro from "../develop/shared/chat-app/_intro.mdx"
-import SetUpDocker from "../develop/shared/chat-app/_set-up-docker.mdx"
+import SetUpCloud from "../develop/shared/chat-app/_set-up-docker.mdx"
```
Comment (michael-redpanda) [inline docs/develop/guide-python-cloud.mdx:19]:
```suggestion
import SetUpCloud from "../develop/shared/chat-app/_set-up-cloud.mdx"
```

**Review by michael-redpanda — COMMENTED** [review]
Getting a weird error when closing the NodeJS example:

```
node:internal/validators:96
      throw new ERR_INVALID_ARG_TYPE(name, 'number', value);
      ^

TypeError [ERR_INVALID_ARG_TYPE]: The "code" argument must be of type number. Received type string ('SIGINT')
    at process.set [as exitCode] (node:internal/bootstrap/node:124:9)
    at process.exit (node:internal/process/per_thread:188:24)
    at process.emit (node:events:511:28) {
  code: 'ERR_INVALID_ARG_TYPE'
}

Node.js v20.1.0
```

And the python one is also still throwing a threading exception:

```
Exception ignored in: <module 'threading' from '/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py'>
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/threading.py", line 1583, in _shutdown
    lock.acquire()
KeyboardInterrupt:
```

The examples work though

**Comment by JakeSCahill** [comment]
> node:internal/validators:96
>       throw new ERR_INVALID_ARG_TYPE(name, 'number', value);
>       ^
> 
> TypeError [ERR_INVALID_ARG_TYPE]: The "code" argument must be of type number. Received type string ('SIGINT')
>     at process.set [as exitCode] (node:internal/bootstrap/node:124:9)
>     at process.exit (node:internal/process/per_thread:188:24)
>     at process.emit (node:events:511:28) {
>   code: 'ERR_INVALID_ARG_TYPE'
> }
> 
> Node.js v20.1.0

For Python, I forgot to add `consumer_t.daemon = True`. It should work now. After one second, it closes.

For Node.js I've updated the closing logic. You may have to press Ctrl + C twice, but you don't get that error.

**Review by michael-redpanda — APPROVED** [review]
LGTM

File: docs/develop/guide-go-cloud.mdx [hunk 5]
```diff
@@ -0,0 +1,390 @@
+---
+title: Build a Chat Room Application with Redpanda Cloud and Go
```
Comment (JakeSCahill) [inline docs/develop/guide-go-cloud.mdx:2]:
```suggestion
title: Build a Chat Room Application with Redpanda Cloud and Golang
```

File: docs/develop/guide-go-cloud.mdx [hunk 6]
```diff
@@ -0,0 +1,390 @@
...
+deployment: Cloud
+contextLinks:
+  - name: 'Docker'
+    to: 'develop/guide-go'
+  - name: 'Cloud'
+    to: 'develop/guide-go-cloud'
+---
+
+<head>
+    <meta name="title" content="Build a Chat Room Application with Redpanda Cloud and Go | Redpanda Docs"/>
```
Comment (JakeSCahill) [inline docs/develop/guide-go-cloud.mdx:13]:
```suggestion
    <meta name="title" content="Build a Chat Room Application with Redpanda Cloud and Golang | Redpanda Docs"/>
```

File: docs/develop/guide-go-cloud.mdx [hunk 7]
```diff
@@ -0,0 +1,390 @@
...
+    <meta name="description" content="Create a basic chat room application with Redpanda Cloud and Kafka clients developed with kafkajs"/>
```
Comment (JakeSCahill) [inline docs/develop/guide-go-cloud.mdx:14]:
```suggestion
    <meta name="description" content="Create a basic chat room application with Redpanda Cloud and Kafka clients developed with franz-go"/>
```

File: docs/develop/guide-python-cloud.mdx [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+contextLinks:
+  - name: 'Docker'
+    to: 'develop/guide-nodejs'
+  - name: 'Cloud'
+    to: 'develop/guide-python-cloud'
+---
+
+<head>
+    <meta name="title" content="Build a Chat Room Application with Redpanda Cloud and Python | Redpanda Docs"/>
+    <meta name="description" content="Create a basic chat room application with Redpanda Cloud and Kafka clients developed with kafkajs"/>
```
Comment (JakeSCahill) [inline docs/develop/guide-python-cloud.mdx:14]:
```suggestion
    <meta name="description" content="Create a basic chat room application with Redpanda Cloud and Kafka clients developed with kafka-python"/>
```

File: docs/develop/guide-go-cloud.mdx [hunk 9]
```diff
@@ -0,0 +1,390 @@
+---
+title: Build a Chat Room Application with Redpanda Cloud and Golang
```
Comment (Feediver1) [inline docs/develop/guide-go-cloud.mdx:2]: Love how this continues the chat room app started in the QS--nice!

File: docs/develop/guide-go-cloud.mdx [hunk 10]
```diff
@@ -0,0 +1,390 @@
...
+  ```
+
+  Output:
+
+  ```
+  TOPIC      STATUS
+  chat-room  OK
+  ```
+
+## Make sure the topic exists on the client side
```
Comment (Feediver1) [inline docs/develop/guide-go-cloud.mdx:100]:
```suggestion
## Confirm that the topic exists on the client side
```

File: docs/develop/guide-go-cloud.mdx [hunk 11]
```diff
@@ -0,0 +1,390 @@
...
+
+## Create a client application
+
+The client needs an application that creates the topic, producer, and consumer and implements the chat logic.
+
+To create a client application, in `chat-room/` create a source file `main.go` with the following content.
+
+Replace the placeholders with the same values that you used in the Redpanda Cloud Quickstart.
+
+```go title="main.go"
```
Comment (Feediver1) [inline docs/develop/guide-go-cloud.mdx:314]: NOTE
These sample files are based on the assumption that you're running the Redpanda Quickstart, where the external port for broker redpanda-0 is set to port 19092. If you're not using the Redpanda Quickstart, make sure that the broker's port is correct for your deployment.

File: docs/develop/guide-go.mdx [hunk 12]
```diff
@@ -0,0 +1,361 @@
...
+  ```
+
+  Output:
+
+  ```
+  TOPIC      STATUS
+  chat-room  OK
+  ```
+
+## Make sure the topic exists on the client side
```
Comment (Feediver1) [inline docs/develop/guide-go.mdx:91]:
```suggestion
## Confirm that the topic exists on the client side
```

File: docs/develop/guide-go.mdx [hunk 13]
```diff
@@ -0,0 +1,361 @@
...
+}
+```
+
+## Create a producer
+
+A client needs a producer to publish `chat-room` topic messages.
+
+To create a producer for the client, in `chat-room/` create a source file `producer.go` with the following content.
+
+```go title="producer.go"
```
Comment (Feediver1) [inline docs/develop/guide-go.mdx:164]: Please update for each app--I assume this is single sourced so just the one update:

NOTE
These sample files are based on the assumption that you're running the Redpanda Quickstart, where the external port for broker redpanda-0 is set to port 19092. If you're not using the Redpanda Quickstart, make sure that the broker's port is correct for your deployment.

File: docs/develop/guide-go.mdx [hunk 14]
```diff
@@ -0,0 +1,361 @@
...
+why the example is using a generated UUID for the group ID. This way, each time you run the application, you see all previous messages.
+
+## Create a client application
+
+The client needs an application that creates the topic, producer, and consumer and implements the chat logic.
+
+To create a client application, in `chat-room/` create a source file `main.go` with the following content.
+
+:::note
+These sample files assume you're running the Redpanda Quickstart, where the external port for broker `redpanda-0` is set to port 19092. If you're not using the Redpanda Quickstart, make sure that the broker's port is correct for your deployment.
```
Comment (Feediver1) [inline docs/develop/guide-go.mdx:282]:
```suggestion
These sample files are based on the assumption that you're running the Redpanda Quickstart, where the external port for broker `redpanda-0` is set to port 19092. If you're not using the Redpanda Quickstart, make sure that the broker's port is correct for your deployment.
```

File: docs/develop/guide-nodejs-cloud.mdx [hunk 15]
```diff
@@ -0,0 +1,367 @@
...
+  ```
+
+  Output:
+
+  ```
+  TOPIC      STATUS
+  chat-room  OK
+  ```
+
+## Make sure the topic exists on the client side
```
Comment (Feediver1) [inline docs/develop/guide-nodejs-cloud.mdx:105]:
```suggestion
## Confirm that the topic exists on the client side
```

File: docs/develop/guide-python-cloud.mdx [hunk 16]
```diff
@@ -0,0 +1,285 @@
...
+  ```
+
+  Output:
+
+  ```
+  TOPIC      STATUS
+  chat-room  OK
+  ```
+
+## Make sure the topic exists on the client side
```
Comment (Feediver1) [inline docs/develop/guide-python-cloud.mdx:90]:
```suggestion
## Confirm that the topic exists on the client side
```

File: docs/develop/guide-python.mdx [hunk 17]
```diff
@@ -0,0 +1,267 @@
...
+  ```
+
+  Output:
+
+  ```
+  TOPIC      STATUS
+  chat-room  OK
+  ```
+
+## Make sure the topic exists on the client side
```
Comment (Feediver1) [inline docs/develop/guide-python.mdx:90]:
```suggestion
## Confirm that the topic exists on the client side
```

File: docs/develop/guide-python.mdx [hunk 18]
```diff
@@ -0,0 +1,267 @@
...
+why the example is using a generated UUID for the group ID.
+
+## Create a client application
+
+The client needs an application that creates the topic, producer, and consumer and implements the chat logic.
+
+To create a client application, in `chat-room/` create a source file `app.py` with the following content.
+
+:::note
+These sample files assume you're running the Redpanda Quickstart, where the external port for broker `redpanda-0` is set to port 19092. If you're not using the Redpanda Quickstart, make sure that the broker's port is correct for your deployment.
```
Comment (Feediver1) [inline docs/develop/guide-python.mdx:194]:
```suggestion
These sample files are based on the assumption that you're running the Redpanda Quickstart, where the external port for broker `redpanda-0` is set to port 19092. If you're not using the Redpanda Quickstart, make sure that the broker's port is correct for your deployment.
```

File: docs/develop/kafka-clients.mdx [hunk 19]
```diff
@@ -25,14 +29,14 @@ The following clients have been validated with Redpanda.
...
 | Rust     | [kafka-rust](https://github.com/kafka-rust/kafka-rust) |
 
-:::tip
-* Use the latest supported version.
-* Other clients, particularly ones based on librdkafka, work with Redpanda, but they have not been validated.
-:::
+Clients that have not been validated by Redpanda Data are still compatible, particularly those based on librdkafka.
+
+If you find a client that is not
+supported, reach out to our team in the community <a href="https://redpanda.com/slack">Slack</a>.
```
Comment (Feediver1) [inline docs/develop/kafka-clients.mdx:35]:
```suggestion
supported, reach out to the Redpanda team in the community <a href="https://redpanda.com/slack">Slack</a>.
```

**Comment by micheleRP** [comment]
Nice, @JakeSCahill! Would you please also update the [Develop index page](https://deploy-preview-1581--redpanda-documentation.netlify.app/docs/develop/ ) to remove the old NodeJS item, and add a nested index page for the pages under [Build a Chat Room Application](https://deploy-preview-1581--redpanda-documentation.netlify.app/docs/develop/code-examples/)?

**Review by Feediver1 — APPROVED** [review]
See comments. Otherwise, LGTM.

**Pull request merged**